cpu/cc2538/radio/cc2538_rf_netdev.c

```diff
@@ -371,7 +371,13 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
                           (CC2538_CORR_VAL_MAX - CC2538_CORR_VAL_MIN);
     }
 
-    RFCORE_SFR_RFST = ISFLUSHRX;
+    /* Check for overflow of the rx fifo */
+    if(RFCORE->XREG_FSMSTAT1bits.FIFOP != 0 &&
+       RFCORE->XREG_FSMSTAT1bits.FIFO == 0)
+    {
+        DEBUG_PRINT("cc2538_rf: RXFIFO Overflow\n");
+        RFCORE_SFR_RFST = ISFLUSHRX;
+    }
```

Comment:
Shouldn't the fifo be flushed on a correct reception as well? (no overflow or underflow)

Reply:
Correction, in the datasheet the is no reference to this being needed, but if the FIFO is not being emptied it could be that the same frame is being read twice because of incorrect ISR handling, and if that is the case the fix here could be preventing the error by having the same frame read twice, which is also not good.

Reply:
I might be missing something here again..

Reply:
My interpretation of the manual is that `rfcore_read_byte` (and thus reading from `RFCORE_SFR_RFDATA`) consumes data from the RXFIFO. So if the entire frame has been read it should no longer be in the buffer.

I can't find a direct mention that this is the behaviour in the manual. The only mention I can find in the manual is that the RXFIFO can be accessed by `RFD` [Section 23.10] (which I assume stands for RFDATA) and that direct accessing (via `RXFIRST_PTR`, `RXLAST_PTR`, and `RXP1_PTR`) doesn't update any pointers.

Reply:
Makes sense, I agree with your interpretation.

Reply:
> 19.4.2.2 Receive FIFO
> The common RX FIFO is a 16-bit-wide, 8-location-deep, first-in first-out memory buffer. Received data
> from the serial interface is stored in the buffer until read out by the CPU, which accesses the read FIFO by
> reading the SSI_DR register.

I guess that can be interpreted as no need to flush after a valid read.

And from the state machine a `flush` doesn't seem to be required on a valid read.

Reply:
I'll test it out!